File: ch36p906XML/p741p758XML/MyPersonParser.m


#import "MyPersonParser.h"
#import "Person.h"

@implementation MyPersonParser

- (void)parser:(NSXMLParser *)parser didStartElement:(NSString *)elementName namespaceURI:(NSString *)namespaceURI qualifiedName:(NSString *)qualifiedName attributes:(NSDictionary *)attributeDict {
    [self makeChild:[MyXMLParserDelegate class] elementName:elementName parser:parser];
}

- (void) finishedChild:(NSString *)s {
    if (!self.person) {
        Person* p = [Person new];
        self.person = p;
    }
    [self.person setValue: s forKey: self.child.name];
}



@end

File: bk2ch23p818xml/ch36p1070xml/MyPersonParser.swift


import UIKit

class MyPersonParser : MyXMLParserDelegate {
    
    var person = Person(firstName: "", lastName: "")
    
    func parser(parser: NSXMLParser, didStartElement elementName: String,
        namespaceURI: String?, qualifiedName qName: String?,
        attributes attributeDict: [NSObject : AnyObject]) {
            self.makeChild(MyXMLParserDelegate.self, elementName: elementName, parser: parser)
    }
    
    override func finishedChild(_ s: String) {
        self.person.setValue(s as NSString, forKey:self.child.name)
    }
    
}
